### PR TITLE
Fix provider to show NotSupported error; Add Get/set/clear content.

### PR DESCRIPTION
**Pull request by BellaLi**

### CODE REVIEW

File: src/Microsoft.PowerShell.SHiPS/SHiPSConstants.cs [hunk 1]
```diff
@@ -26,6 +26,20 @@ internal static class Constants
             "Resolve-Path",
             "Get-Item",
             "Test-Path"
+        };
+
+        // Although we don't support these commands, we do handle their error messages.
+        internal static string[] NotSupportedCommands = {
```
Comment (jianyunt) [inline src/Microsoft.PowerShell.SHiPS/SHiPSConstants.cs]: should we combine Supported and Not supported commands to one providercommands?

File: src/p2f/src/CodeOwls.PowerShell/CodeOwls.PowerShell.Provider/Provider.cs [hunk 2]
```diff
@@ -1541,14 +1542,15 @@ private IContentWriter DoGetContentWriter(string path)
...
-            return GetContentWriter(path, factories.First(a => a is ISetItemContent));
+            return GetContentWriter(path, factories.FirstOrDefault(a => a is ISetItemContent));
         }
 
         private IContentWriter GetContentWriter(string path, IPathNode pathNode)
         {
             var getContentWriter = pathNode as ISetItemContent;
             if (null == getContentWriter)
             {
+                WriteCmdletNotSupportedAtNodeError(path, ProviderCmdlet.SetContent, SetContentNotSupportedErrorId);
```
Comment (jianyunt) [inline src/p2f/src/CodeOwls.PowerShell/CodeOwls.PowerShell.Provider/Provider.cs]: Please add test cases for happy path, for example,  Set-Content  or New-Item  /$home/foo/t.txt should work  and get-content ~/foo/t.txt, is expected to work.

File: src/Microsoft.PowerShell.SHiPS/Utility/SHiPSExtensions.cs [hunk 3]
```diff
@@ -220,10 +220,12 @@ internal static bool UsingDynamicParameter(this Dictionary<string, object> bound
             return usingDynamicParameter || filter;
         }
 
-        internal static bool IsProviderSupportedCommand(this IProviderContext context)
+        internal static bool IsProviderDefinedCommand(this IProviderContext context)
         {
             var myInvocationInfo = GetMyInvocationInfo(context.CmdletProvider);
-            return myInvocationInfo?.MyCommand != null && Constants.SupportedCommands.ContainsIgnoreCase(myInvocationInfo.MyCommand.Name);
+            return myInvocationInfo?.MyCommand != null 
+                && (Constants.SupportedCommands.ContainsIgnoreCase(myInvocationInfo.MyCommand.Name) || Constants.NotSupportedCommands.ContainsIgnoreCase(myInvocationInfo.MyCommand.Name)
```
Comment (jianyunt) [inline src/Microsoft.PowerShell.SHiPS/Utility/SHiPSExtensions.cs]: >Constants.SupportedCo [](start = 20, length = 21)

Supported and NotSupported will be merged to one. please

File: src/Microsoft.PowerShell.SHiPS/SHiPSConstants.cs (hunk 1 above)
Comment (BellaLi) [reply src/Microsoft.PowerShell.SHiPS/SHiPSConstants.cs]: We can, if there is no other usage of the Supported list, now and future.

File: src/p2f/src/CodeOwls.PowerShell/CodeOwls.PowerShell.Provider/Provider.cs (hunk 2 above)
Comment (BellaLi) [reply src/p2f/src/CodeOwls.PowerShell/CodeOwls.PowerShell.Provider/Provider.cs]: Will do. Thanks!

File: test/automation/navigation.tests.ps1 [hunk 4]
```diff
@@ -1209,7 +1209,135 @@ AfterEach{
...
+        cd $testpath
+       }
+
+AfterEach{
+        cd $home
+        $a=Get-PSDrive -PSProvider SHiPS
+        $a | % {remove-psdrive $_.Name -ErrorAction Ignore}
+
+        If(Test-Path $script:homePath) {Remove-Item -Path $script:homePath -force -Recurse -ErrorAction Ignore}
+
```
Comment (mirichmo) [inline test/automation/navigation.tests.ps1]: My comments from the first PR still apply to these tests.

Comment (BellaLi) [reply test/automation/navigation.tests.ps1]: Fixed.

**Comment by jianyunt** [comment]
@BellaLi  please fix the checkin test failures 
 https://ci.appveyor.com/project/PowerShell/ships/build/1.1.3.90/job/b2o9habi4vochq66
C:\projects\ships\test\automation\navigation.tests.ps1: line 1335
     --- have you changed to "not support" instead of "does not support"?
C:\projects\ships\test\automation\navigation.tests.ps1: line 1249.
     --- it looks like that the same error throws multiple times. So $ev is a collection. "-contains" would do.

File: src/Microsoft.PowerShell.SHiPS/SHiPSProvider.cs [hunk 5]
```diff
@@ -15,7 +15,8 @@ namespace Microsoft.PowerShell.SHiPS
     /// allows for stateless namespace navigation of a system you are modeling.
     /// </summary>
     [CmdletProvider(SHiPSProvider.ProviderName, ProviderCapabilities.Filter | ProviderCapabilities.ShouldProcess)]
-    public class SHiPSProvider : CodeOwls.PowerShell.Provider.Provider
+    public class SHiPSProvider : CodeOwls.PowerShell.Provider.Provider,
+        IContentCmdletProvider
```
Comment (jianyunt) [inline src/Microsoft.PowerShell.SHiPS/SHiPSProvider.cs]: Are you going to remove it as we discussed it before?

**Comment by jianyunt** [comment]
This PR and https://github.com/PowerShell/SHiPS/pull/105 are dealing with the same thing or different?

**Comment by BellaLi** [comment]
The previous PR 105 was accidentally checked in. I'll abandon this PR and create a clean one.

**Pull request closed without merging**